processing/src/main/java/org/apache/carbondata/processing/surrogatekeysgenerator/csvbased/CarbonCSVBasedSeqGenStep.java

```diff
@@ -1133,6 +1149,9 @@ else if(isComplexTypeColumn[j]) {
                       .getDirectDictionaryGenerator(details.getColumnType());
               surrogateKeyForHrrchy[0] =
                   directDictionaryGenerator1.generateDirectSurrogateKey(tuple);
+              if (badRecordsLoggerEnable && surrogateKeyForHrrchy[0] == 1) {
+                surrogateKeyForHrrchy[0] = -1;
```

Comment:
Replace -1 and 1 with proper identifiers from CarbonCommonConstants to give proper meaning and improve code readability